webapp/src/main/java/com/box/l10n/mojito/service/ai/openai/OpenAILLMService.java

```diff
@@ -56,31 +58,38 @@ public class OpenAILLMService implements LLMService {
   boolean persistResults;
 
   @Timed("OpenAILLMService.executeAIChecks")
-  public AICheckResponse executeAIChecks(AICheckRequest AICheckRequest) {
+  public AICheckResponse executeAIChecks(AICheckRequest aiCheckRequest) {
 
     logger.debug("Executing OpenAI string checks.");
-    Repository repository = repositoryRepository.findByName(AICheckRequest.getRepositoryName());
+    Repository repository = repositoryRepository.findByName(aiCheckRequest.getRepositoryName());
 
     if (repository == null) {
-      logger.error("Repository not found: {}", AICheckRequest.getRepositoryName());
-      throw new AIException("Repository not found: " + AICheckRequest.getRepositoryName());
+      logger.error("Repository not found: {}", aiCheckRequest.getRepositoryName());
+      throw new AIException("Repository not found: " + aiCheckRequest.getRepositoryName());
     }
 
     List<AIPrompt> prompts =
         LLMPromptService.getPromptsByRepositoryAndPromptType(repository, SOURCE_STRING_CHECKER);
 
+    Set<String> sourceStrings = new HashSet<>();
     Map<String, List<AICheckResult>> results = new HashMap<>();
-    AICheckRequest.getTextUnits()
+    aiCheckRequest
+        .getTextUnits()
         .forEach(
             textUnit -> {
-              List<AICheckResult> AICheckResults = checkString(textUnit, prompts, repository);
-              results.put(textUnit.getSource(), AICheckResults);
+              // Only check each source string once, this prevents duplicate checks for the same
```

Comment:
While it would mean going through the list twice, building the unique set separate might make this code easier to read inside the stream.

E.g. something like:
```
Map<String, TextUnit> uniqueTextUnits = aiCheckRequest.getTextUnits().stream()  
    .collect(Collectors.toMap(  
        TextUnit::getSource,   
        textUnit -> textUnit,   
        (existing, replacement) -> existing,  
        LinkedHashMap::new  
    ));  

uniqueTextUnits.values().forEach(textUnit -> 
// same as before
```